pkg/kubectl/cmd/util/shortcut_restmapper.go

```diff
@@ -40,13 +42,20 @@ func NewShortcutExpander(delegate meta.RESTMapper, client discovery.DiscoveryInt
 	return ShortcutExpander{All: userResources, RESTMapper: delegate, discoveryClient: client}
 }
 
-func (e ShortcutExpander) getAll() []schema.GroupResource {
+func (e ShortcutExpander) getServerRes(groupVersion string) ([]*metav1.APIResourceList, error) {
```

Comment:
pinned to what ? to `ShortcutExpander` ? - do you suspect that the list obtained from the server can change over the lifetime of the `kubectl` process ?